x-pack/plugins/global_search_bar/public/components/search_bar.tsx

```diff
@@ -230,16 +208,17 @@ export const SearchBar: FC<SearchBarProps> = ({
       // errors in tracking should not prevent selection behavior
       try {
         if (type === 'application') {
-          const key = selected.keys ?? 'unknown';
```

Comment:
This was the bug that caused the application to be identified as `unknown` for every execution of this code.